### ISSUE TITLE
Non-Sandboxed Moonshine Build

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#537]
@JustinProminic requested that we create a version of Moonshine without the sandbox, which we can distribute from the website.  We hope that this will let us avoid issues like those reported in #493.  For example:
- We should be able to install SDKs outside of ~/Downloads.  In particular, this caused trouble in the sandbox with the Flex, Royale, Ant, and Maven SDKs
- We should be able to access shared global directories like ~/.m2, the local Maven repository.

This may also allow us to do other actions that were restricted by the sandbox, like Camera access (#479).  Even if we still need entitlements, we should be able to add them freely in the non-sandbox version of Moonshine.

Let us know if there are unexpected complications for this change, and we will reevaluate the priority.

**Comment by rat-moonshine** [comment]
Separate entitlement now builds the non-sandbox version; following changes have done to the entitlement file:

```
// before
<key>com.apple.security.app-sandbox</key>
<true/>
<key>com.apple.security.files.downloads.read-only</key>
<true/>

// now
<key>com.apple.security.app-sandbox</key>
<false/>
<key>com.apple.security.files.downloads.read-write</key>
<true/>
```

Build script updated to build the non-sandbox version with correct entitlement. "Access Manager" removed in the version; Regular `File` API will now be in use instead of bookmarking ANE (bookmarking ANE requires to be run in sandbox). 

In initial tests, SDK resides in non-Downloads section worked good in project compilation. 

Please, report if any other parts we requires to adjust on this course.

**Comment by JoelProminic** [comment]
I confirmed I can use Ant from Homebrew (#493) and can run a Camera project (#479) from OSX Yosemite.  I'll plan to test with a High Sierra Mac as well before release.

One potential problem is that the non-Sandbox application uses a different configuration directory, so users may notice that their projects and settings change after they switch to the non-sandbox version of the application.

**Comment by rat-moonshine** [comment]
To keep a note, in my tests on El Capitan, I found difficulty creating new projects. However, this worked good in Joel's test. 

I would like to investigate on this and find a reason why this different between our systems.

**Comment by rat-moonshine** [comment]
- Fix: Bookmark path existence check causing trouble during project creation when bookmark collection is empty
- About screen shall show "(App Store Version)" label in case of sandbox/App-store version. Windows and non-sandbox-OSX version will not accompany this label

**Comment by piotrzarzycki21** [comment]
I propose once you push to the appstore version of Moonshine, add on the website non sandbox version as well for download. @rat-moonshine Will you be able to do that ? If yes please close this issue and #493

**Comment by rat-moonshine** [comment]
Generally we updates installers to Moonshine website once the App Store version pass approval. I think we shall remain to that course. Yes, the plan is to release the non-sandbox version from v2.2.0 only, as we haven't noticed yet any problem towards the purpose of using non-sandbox version.

You can also keep using the non-sandbox version until then if you see any problem, Piotr!

**Comment by piotrzarzycki21** [comment]
> Generally we updates installers to Moonshine website once the App Store version pass approval.

Yes and I'm not asking to do that stuff differently. I'm saying that as an additional link on the website should appear non-sandbox version with clear message. - ONCE we release AppStore version - Unless there is different desire from Justin.

**Comment by rat-moonshine** [comment]
I see we don't have a direct download link anymore in newly constructed download page (http://moonshine-ide.com/download/), except in archive downloads section. So we probably require to include direct download links for both Windows and OSX on new download page.

**Comment by piotrzarzycki21** [comment]
I'm closing this one since we have released version 2.2.0
